Modules/Numerics/Optimizersv4/include/itkQuasiNewtonOptimizerv4.h

```diff
@@ -151,8 +151,11 @@ class ITK_TEMPLATE_EXPORT QuasiNewtonOptimizerv4Template
   /** The Hessian with local support */
   HessianArrayType m_HessianArray;
 
-  /** Valid flag for the Quasi-Newton steps */
-  std::vector<bool> m_NewtonStepValidFlags;
+  /** Valid flag for the Quasi-Newton steps.
+   * NB: although semantically boolean, vector<bool> is not thread safe due to the possibility of multiple bits being
+   * packed together in the same memory location.
+   */
+  std::vector<int> m_NewtonStepValidFlags;
```

Comment:
Good catch @seanm !

Please consider ` std::vector<uint8_t>`, as you did before, with pull request #3020 commit b66133ab53369bdf265348c0c31cfa6451b53934

BTW, if the use of `std::vector<"semantically boolean">` appears more often in ITK, I would suggest adding a type alias, for example `using Bool = uint8_t`. So that it can be documented in one place.